Style Leaflet previews

bronzehedwick opened the pull request:
Initial pass at styling the map previews. I made some light design decisions, and generally tried to keep everything inside the preview and at a reasonably small size. I added a new Leaflet Overrides CSS file to make the tweaks, that is only loaded when the Leaflet map is rendered.

There's more work left to do in the task that will be coming soon, but wanted to slice this chunk out for review now.

Open to any feedback. Thanks!

Re #343

bronzehedwick committed:
Style Leaflet previews

michael-wynne-wsu commented:
A great start! Looks like you focused on the DH item previews, right? (I didn't check PDFs, video upload, or embeds this time)

Thoughts/notes/questions - I'm always up for discussion on these of course. 
1) I don't think we need the extra white space and bottom grey line below the title.
![Screenshot 2025-03-17 at 11 32 17 AM](https://github.com/user-attachments/assets/0efd349c-3914-45f5-a6b6-01ce51e36161)

2) Overall the previews feel a bit small - any opinion on bumping the title text size up a little? I'm thinking we may want to set minimum preview dimensions? 

3) Related, when the title is short it really shrinks down the size of the preview (am I correct that the width is based off title length?), and with that, how is the height calculated? Examples are especially notable with the audio player that can be so small it completely collapses.
![Screenshot 2025-03-17 at 10 12 53 AM](https://github.com/user-attachments/assets/6d72a6e5-aee4-47cb-9ab0-6426c0e45fd5)

4) Vimeo seems to love including excess top and bottom padding. This is most obvious with a smaller preview, but it's something we see across our sites and elsewhere as well.
![Screenshot 2025-03-17 at 10 13 53 AM](https://github.com/user-attachments/assets/7cb0d9e9-65a4-45d9-9b74-b7352cb64fbc)
![Screenshot 2025-03-17 at 10 14 46 AM](https://github.com/user-attachments/assets/98cff488-0888-4059-bb9a-aede36777020)

5) The layer button in the top right is exposing I think a per-item filter? Not sure if we want to use that for anything in particular or just get rid of it. I can think of some other tools that would be neat in the map but they're probably future tickets.
![Screenshot 2025-03-17 at 10 14 07 AM](https://github.com/user-attachments/assets/d4f1a327-b0cc-4afb-9103-4c0cf8037464)

bronzehedwick commented:
Thanks @michael-wynne-wsu! Responses below.

> A great start! Looks like you focused on the DH item previews, right? (I didn't check PDFs, video upload, or embeds this time)

Right, this PR doesn't address PDF preview generation, although it should work for direct video uploads or generic embeds.

> I don't think we need the extra white space and bottom grey line below the title.

Sure thing, I'll remove that in this context.

> Overall the previews feel a bit small - any opinion on bumping the title text size up a little? I'm thinking we may want to set minimum preview dimensions?

Sure, no issue bumping size. I left it as the default vertical card styles specifically to gauge feedback. I'll play with this a bit, but I'm thinking getting closer to the horizontal card font size.

> Related, when the title is short it really shrinks down the size of the preview (am I correct that the width is based off title length?), and with that, how is the height calculated? Examples are especially notable with the audio player that can be so small it completely collapses.

Yes, the Leaflet library inserts inline styles to make the preview roughly the width of the content. I can play with adding some overrides for this, although for instances similar in the past, that can sometimes mess up the libraries geometry calculations, so I may report something along those lines.

> Vimeo seems to love including excess top and bottom padding. This is most obvious with a smaller preview, but it's something we see across our sites and elsewhere as well.

Gotcha. I'll test Vimeo specifically and see if I can tame that padding.

> The layer button in the top right is exposing I think a per-item filter? Not sure if we want to use that for anything in particular or just get rid of it. I can think of some other tools that would be neat in the map but they're probably future tickets.

I can look into disabling this for now, and then we can re-evaluate if/when a use is found.

bronzehedwick committed:
Remove map vertical card border

michael-wynne-wsu commented:
@bronzehedwick 

> Sure, no issue bumping size. I left it as the default vertical card styles specifically to gauge feedback. I'll play with this a bit, but I'm thinking getting closer to the horizontal card font size.
Great! I'm not sure what the horizontal card size it, but we're definitely not restricted to the vertical style, if you think the horizontal layout (or anything else) is worth using/exploring too.

> Yes, the Leaflet library inserts inline styles to make the preview roughly the width of the content. I can play with adding some overrides for this, although for instances similar in the past, that can sometimes mess up the libraries geometry calculations, so I may report something along those lines.
Increasing the font size as above might be enough to address this as well. I do think keeping the audio player from shrinking so far it collapses would be good though.

> I can look into disabling this for now, and then we can re-evaluate if/when a use is found.
Perfect, I suspect we won't need it.

bronzehedwick committed:
Increase map preview font size

bronzehedwick commented:
@michael-wynne-wsu Looks like Leaflet has min/max preview widths, which can be configured per instance. The `/digital-heritage` one has the max at 300, and the min at 50. I'll play with increasing those.

bronzehedwick committed:
Increase the min/max preview width

bronzehedwick committed:
Give all media embeds a black background

This clears up confusion around Vimeo's display; if a video's aspect
ratio is too short for the container, the accepted visual is black bars,
while Vimeo uses transparent ones, which ends up looking like white
space.

bronzehedwick committed:
Hide layer controls in CSS

Not a permanent solution.

bronzehedwick commented:
Pushed fixes for what you brought up above, @michael-wynne-wsu. LMK if I should adjust!

michael-wynne-wsu commented:
@bronzehedwick this is looking great.

Agreed that the black backgrounds for both remote video sources is an improvement for consistency.

Increased font size, removing the vertical borders, removing the extraneous layer control all looks good.

The only things that still feel off right now are:
1) Audio player not working, but you noted that already.
2) For the video (and to a lesser extent audio), the previews are still just too small (at least at the size on my displays) with a short title to be really accessible. Any chance we could bump minimum width to ~300px? See below for unaltered (~130px) vs 300px wide.

![Screenshot 2025-03-17 at 2 38 54 PM](https://github.com/user-attachments/assets/71557003-4f8a-4f79-a99b-8730508a4e1b)

![Screenshot 2025-03-17 at 2 38 41 PM](https://github.com/user-attachments/assets/122bfe29-2264-42ca-b076-fdbed173e826)

bronzehedwick committed:
Increase preview width to min 300

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr725-zxqtwpxozvcfq7zslibrm9euqat7iito.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67d8545ee6cc8e4e840f338b

bronzehedwick commented:
Hey @michael-wynne-wsu, thanks for the feedback!

> Audio player not working, but you noted that already.

Yes, still working on that one. I'd like to slice that work out to a separate PR, as it is unrelated to these style changes, and is taking longer than I'd like. I can speak more to it in today's meeting.

> Any chance we could bump minimum width to ~300px? See below for unaltered (~130px) vs 300px wide.

Done! I've changed the minimum preview width to be `300px`, and increased the maximum to `400px`. Let me know how that feels.

michael-wynne-wsu commented:
@bronzehedwick agreed re: audio player, and I stubbed out #727 for that.

I'm not seeing the updated minimum card width reflected in the tugboat build yet

michael-wynne-wsu commented:
@bronzehedwick Ahhh I was looking at /browse, NOT /digital-heritage, so YES it's looking good at /digital-heritage!

bronzehedwick commented:
@michael-wynne-wsu Thanks! Should have specified, and I'll go ahead and find and update the other views/content types with the new min/max.

michael-wynne-wsu commented:
@bronzehedwick Do you want to merge this PR as is or are you planning to include more in it?

bronzehedwick commented:
@michael-wynne-wsu I'd like to hold off merging for a moment while I migrate everything over to use the Map Display Mode instead of the List Display Mode; turns out keeping them the same was causing the List view to regress!

michael-wynne-wsu commented:
@bronzehedwick makes sense!

bronzehedwick committed:
Switch to map browse display mode for maps

bronzehedwick committed:
Move Leaflet styles to global

Library attach stopped working. Not sure what's going on, but it's a
small optimization, so saving as a follow up task.

bronzehedwick commented:
Okay @michael-wynne-wsu, pushed the fixes for the Map Display Mode! Should probably be looked at one more time, and then we can merge.

Note that I'll handle the `/browse` and other pages once Nate follows up on that PHP error, as well as the other unhandled tasks in the main task list, in follow up PRs.

michael-wynne-wsu commented:
@bronzehedwick the previews are looking great - I think we can merge this one.
